lessons/git/git-collaboration-2in1/index.md

```diff
@@ -0,0 +1,291 @@
+V podstatě mu musíme dát příkaz, aby zmrazil současný stav repozitáře, tzv. vytvořit *revizi*.
+Pak můžeme dělat další změny, ale už vždy budeme schopni se k této revizi vrátit.
+
+Přidejte do repozitáře soubor `vase_jmeno.txt`, např. tedy `magdalena_rettigova.txt`
+a do něj napište svoje současné povolání nebo cokoliv jiného.
+Zkontrolujte současný stav repozitář pomocí `git status`: Git oznámí, že v adresáři je soubor, o kterém ještě „neví“. 
+
+    $ git status
+    On branch master
+    Your branch is up-to-date with 'origin/master'.
+    Untracked files:
+      (use "git add <file>..." to include in what will be committed)
+
+          magdalena_rettigova.txt
+
+    nothing added to commit but untracked files present (use "git add" to track)
+
+U každého nového souboru musíme Gitu říct, že chceme jeho obsah sledovat. Proveďte to se svým souborem:
+
+    $ git add magdalena_rettigova.txt
+
+a znovu zkontroluj stav repozitáře:
+
+    $ git status
+    On branch master
+    Your branch is up-to-date with 'origin/master'.
+    Changes to be committed:
+      (use "git reset HEAD <file>..." to unstage)
+
+            new file:   magdalena_rettigova.txt
+
+To, co je zelené („changes to be committed“), se přidá do další revize (angl. commit), kterou vytvoříme. Pojďme tedy vytvořit revizi: 
+
+    $ git commit
+
+Po zadání tohoto příkazu se otevře editor, do kterého je nutné napsat výstižný popis změn, 
+abyste vy i ostatní programátoři věděli, co tahle revize obsahuje za změny.
+Např. tedy `Přidáno mé jméno a povolání`. Předvyplněné řádky začínající # můžeme nechat být
+(nebo vymazat, podle chuti – Git je ignoruje).
+Pak soubor uložíme a zavřeme editor. 
+    
+    $ git status
+    On branch master
+    Your branch is ahead of 'origin/master' by 1 commit.
+      (use "git push" to publish your local commits)
+    nothing to commit, working tree clean
+
+Pro lepší pochopení, co dělají jednotlivé příkazy, a v jakém stavu můžou být soubory/změny, přikládáme tento diagram: 
+
+![Git workflow]({{ static('diagram.png') }})
+
+#### Log (`git log`)
+Teď, když za sebou máme první revizi, podívejme se, jak vypadá historie repozitáře, do kterého chceme přispívat.
+K tomu můžeme použít příkaz `git log`, jako první uvidíme naši poslední změnu a další jsou změny od jiných autorů,
+na které navazujeme.
+
+    $ git log
+    commit 0bdfbb2a2398fea179395a8dd303e9f672ef4dca
+    Author: Magdalena Dobromila <magdalena@rettigova.cz>
+    Date:   Tue Mar 28 19:27:08 2017 +0200
+
+        Přidáno mé jméno a povolání
+
+    commit 0f305972803131cb6c8637359fee8ede3005bba6
+    Merge: effa89b 175f4cc
+    Author: Oskar Hollmann <oskar@hollmann.me>
+    Date:   Tue Nov 8 20:17:00 2016 +0100
+
+        Merge pull request #1 from muzikovam/master
+        
+        Tady mas jmeno
+
+Příkaz `git log` má mnoho přepínačů, přehlednější historii změn můžeme získat pomocí `git log --oneline --graph --decorate`.
```

Comment:
gitk jsem vyhodil, protože byl na Linuxu často problém s tím, že nebyl nainstalovaným navíc ten log s přepínači je moc pěkný, také jim to chci ukázat v interfacu GitHubu, takže není důvod mít tu další GUI.